modules/xmpp/xmpp.js

```diff
@@ -278,6 +278,11 @@ export default class XMPP extends Listenable {
 
         this.eventEmitter.emit(XMPPEvents.CONNECTION_STATUS_CHANGED, credentials, status, msg);
         if (status === Strophe.Status.CONNECTED || status === Strophe.Status.ATTACHED) {
+            // once connected or attached we no longer need this handle, drop it if it exist
+            if (this._sysMessageHandler) {
```

Comment:
if it checks for a truthy value here, maybe the it should be cleared? or if it's safe to call it multiple times then do it always/unconditionally?

Reply:
Nope, I don't see such check in strophe.